python/pyarrow/_dataset.pyx

```diff
@@ -929,7 +929,8 @@ class RowGroupInfo:
         def name_stats(i):
             col = self.metadata.column(i)
 
-            if not col.statistics.has_min_max:
+            stats = col.statistics
+            if stats is None or not stats.has_min_max:
```

Comment:
Another option would be to check `col.is_stats_set`, but that results in `col.statistics` being None, so this is just as good